src/plugins/index_pattern_field_editor/public/components/field_editor_flyout_content.tsx

```diff
@@ -101,6 +111,14 @@ const FieldEditorFlyoutContentComponent = ({
     }
   }, [onSave, submit]);
 
+  const namesNotAllowed = indexPattern.fields.map((fld) => fld.name);
+  const existingConcreteFields = indexPattern.fields
+    .filter((fld) => fld.isMapped)
+    .map((fld) => ({
+      name: fld.name,
+      type: fld.esTypes![0]!,
```

Comment:
@alisonelizabeth I wanted to point this out to you. The painless autocomplete needs a single type value although index pattern fields can have more than one type. We should address how this is handled to avoid running into problems.